### PR TITLE
chore: removing dependance on mainQuery from conversion screen

### PR DESCRIPTION
**Pull request by nicolasburtey**
also removing some duplicated states

### CODE REVIEW

File: app/graphql/generated.ts [hunk 1]
```diff
@@ -61,12 +61,14 @@ export type Scalars = {
 }
 
 export type Account = {
+  readonly btcWallet?: Maybe<Wallet>
```
Comment (bodymindarts) [inline app/graphql/generated.ts:64]: isn't `?` redundant when using `Maybe` type?

Comment (nicolasburtey) [reply app/graphql/generated.ts:64]: Don't know for sure. But it's a generated files

File: app/graphql/cache.ts [hunk 2]
```diff
@@ -29,5 +30,33 @@ export const cache = new InMemoryCache({
         },
       },
     },
+    ConsumerAccount: {
```
Comment (daviroo) [inline app/graphql/cache.ts:33]: When I attempted this last time I wrote a script to generate the possible types from the schema on the server https://github.com/GaloyMoney/galoy-mobile/compare/main...refactor-transactions-screen#diff-d3e97a63b7fd33227d3577ef3a18ec8793bf860730677e9f7f3382be13d0cc8f.

I was then able to pass in the generated file to the possibleTypes config - https://github.com/GaloyMoney/galoy-mobile/compare/main...refactor-transactions-screen#diff-05ff1f30a72721bd2ff7481a4e51b12592d8d7ed89c5ac459c25c9c9d2505314

**Review by daviroo — CHANGES_REQUESTED** [review]

**Comment by daviroo** [comment]
Since we have polymorphic relationships in our graphql schema I think we need to give the client a possible types map (I left a comment in the code as well).  There are more details in the docs https://www.apollographql.com/docs/react/data/fragments/#using-fragments-with-unions-and-interfaces

**Review by daviroo — APPROVED** [review]

File: app/graphql/cache.ts [hunk 3]
```diff
@@ -1,33 +1,104 @@
...
+            read: (value) => value ?? false,
+          },
         },
-        hiddenBalanceToolTip: {
-          read: (value) => value ?? false,
+      },
+      ConsumerAccount: {
+        fields: {
+          usdWallet: {
+            read: (_, { readField }) => {
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:40]:
```suggestion
            read: (_, { readField, cache }) => {
```

File: app/graphql/cache.ts [hunk 4]
```diff
@@ -1,33 +1,104 @@
...
+              const wallets = readField("wallets")
+              if (Array.isArray(wallets)) {
+                for (const w of wallets) {
+                  try {
+                    const wallet = client?.readFragment({
```
Comment (UncleSamtoshi) [inline app/graphql/cache.ts:45]:
```suggestion
                    const wallet = cache.readFragment({
```

**Review by UncleSamtoshi — CHANGES_REQUESTED** [review]
You can use the `cache` argument instead of having to set the client.

File: app/graphql/cache.ts (hunk 3 above)
Comment (nicolasburtey) [reply app/graphql/cache.ts:40]: oh awesome. thanks! I'll refactor

**Review by UncleSamtoshi — APPROVED** [review]

**Pull request merged**